cmd/client-s3.go

```diff
@@ -1326,40 +1363,23 @@ func (c *S3Client) Stat(isIncomplete, isFetchMeta, isPreserve bool, sse encrypt.
 	//     - /path/to/empty_directory
 	//     - /path/to/empty_directory/
 
-	nonRecursive := false
-	objectMetadata := &ClientContent{}
-
-	// Prefix to pass to minio-go listing in order to fetch a given object/directory
-	prefix := strings.TrimRight(object, string(c.targetURL.Separator))
-
-	// If the request is for incomplete upload stat, handle it here.
-	if isIncomplete {
-		for objectMultipartInfo := range c.api.ListIncompleteUploads(bucket, prefix, nonRecursive, nil) {
-			if objectMultipartInfo.Err != nil {
-				return nil, probe.NewError(objectMultipartInfo.Err)
-			}
+	opts := minio.StatObjectOptions{}
+	opts.ServerSideEncryption = sse
 
-			if objectMultipartInfo.Key == object {
-				objectMetadata.URL = *c.targetURL
-				objectMetadata.Time = objectMultipartInfo.Initiated
-				objectMetadata.Size = objectMultipartInfo.Size
-				objectMetadata.Type = os.FileMode(0664)
-				objectMetadata.Metadata = map[string]string{}
-				return objectMetadata, nil
-			}
+	ctnt, err := c.getObjectStat(bucket, object, opts)
```

Comment:
Shouldn't we do the object stat only if the object name is not suffixed with the slash separator?

Reply:
> Shouldn't we do the object stat only if the object name is not suffixed with the slash separator?

Yes we can optimize it